babel/numbers.py

```diff
@@ -930,12 +940,12 @@ def parse_decimal(string: str, locale: Locale | str | None = LC_NUMERIC, strict:
     decimal_symbol = get_decimal_symbol(locale)
 
     if not strict and (
-        group_symbol == '\xa0' and  # if the grouping symbol is U+00A0 NO-BREAK SPACE,
+        re.match(r"\s", group_symbol) and  # if the grouping symbol is a kind of space,
         group_symbol not in string and  # and the string to be parsed does not contain it,
-        ' ' in string  # but it does contain a space instead,
+        re.search(r"\s", string)  # but it does contain any other kind of space instead,
```

Comment:
Same perf stuff here. If you end up using a cached helper function, it can well be shared between these two functions.